### PR TITLE
Refactoring in daemon/discovery.go

### PR DESCRIPTION
**Pull request by LK4D4**
Replace time.Sleep with time.Tick and remove unnecessary var block

### CODE REVIEW

File: daemon/discovery.go [hunk 1]
```diff
@@ -39,10 +36,10 @@ func initDiscovery(backend, address string, clusterOpts map[string]string) (disc
 // address. The function never returns, as registration against the backend comes with a TTL and
 // requires regular heartbeats.
 func registrationLoop(discoveryBackend discovery.Backend, address string) {
-	for {
+	ticker := time.Tick(defaultDiscoveryHeartbeat)
```
Comment (runcom) [inline daemon/discovery.go]: can't it be just `for range time.Tick(...`?

**Comment by runcom** [comment]
small nit, LGTM otherwise

**Comment by LK4D4** [comment]
@runcom changed

**Comment by abronan** [comment]
LGTM

**Comment by MHBauer** [comment]
Technically this changes the function. It waits 20 seconds before registering, rather than registering and waiting 20 seconds.

**Comment by LK4D4** [comment]
@MHBauer You're right, probably shouldn't be like this.

**Comment by duglin** [comment]
golang needs a do-while  :-)

**Comment by LK4D4** [comment]
@runcom @abronan @MHBauer I changed code to match old behavior.

**Comment by runcom** [comment]
LGTM (missed that)

**Comment by abronan** [comment]
thanks for catching this @MHBauer, missed that too. LGTM.

**Comment by MHBauer** [comment]
Looks functionally equivalent to me now.
@duglin agree on do-while.

**Pull request merged**